diff_drive_controller/include/diff_drive_controller/diff_drive_controller.h

```diff
@@ -229,10 +229,11 @@ namespace diff_drive_controller{
 
     realtime_tools::RealtimeBuffer<DynamicParams> dynamic_params_;
 
-    /// Dynamic Reconfigure server
     typedef dynamic_reconfigure::Server<DiffDriveControllerConfig> ReconfigureServer;
-    
-    std::shared_ptr<ReconfigureServer> dyn_reconf_server_;
+    /// To avoid Dynamic Reconfigure Server warning
+    boost::recursive_mutex dyn_reconf_server_mutex_;
+    /// Dynamic Reconfigure server
+    boost::shared_ptr<ReconfigureServer> dyn_reconf_server_;
```

Comment:
As mentioned in https://github.com/ros-controls/ros_controllers/pull/479#discussion_r404443977, this `boost::shared_ptr` still needs to be switched back to `std::shared_ptr`.

Also, bit of a nit-pick: Even though it isn't perfect Doxygen, let's keep with this file's grouping comment pattern, and place the `typedef ... T` and `shared_ptr<T>` together before the mutex:

```
    /// Dynamic Reconfigure server:
    typedef dynamic_reconfigure::Server<DiffDriveControllerConfig> ReconfigureServer;
    std::shared_ptr<ReconfigureServer> dyn_reconf_server_;
    boost::recursive_mutex dyn_reconf_server_mutex_;
```